src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java

```diff
@@ -29,33 +32,44 @@
 import static org.junit.jupiter.api.Assertions.fail;
 
 @DatabaseTest
+@Execution(ExecutionMode.SAME_THREAD)
 class DBMSProcessorTest {
 
-    @ParameterizedTest
-    @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testCheckBaseIntegrity(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
+    private DBMSConnection dbmsConnection;
+    private DBMSProcessor dbmsProcessor;
+    private DBMSType dbmsType;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        this.dbmsType = TestManager.getDBMSTypeTestParameter();
+        this.dbmsConnection = TestConnector.getTestDBMSConnection(dbmsType);
+        this.dbmsProcessor = DBMSProcessor.getProcessorInstance(TestConnector.getTestDBMSConnection(dbmsType));
+        TestManager.clearTables(this.dbmsConnection);
+    }
+
+    @AfterEach
+    public void clear() throws SQLException {
+        this.dbmsConnection.getConnection().close();
+    }
+
+    @Test
+    void testCheckBaseIntegrity() throws SQLException {
         dbmsProcessor.setupSharedDatabase();
         assertTrue(dbmsProcessor.checkBaseIntegrity());
-        clear(dbmsConnection);
+        TestManager.clearTables(this.dbmsConnection);
         assertFalse(dbmsProcessor.checkBaseIntegrity());
-
-        clear(dbmsConnection);
     }
 
-    @ParameterizedTest
-    @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testSetUpSharedDatabase(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
+    @Test
+    void testSetUpSharedDatabase() throws SQLException {
         dbmsProcessor.setupSharedDatabase();
-        clear(dbmsConnection);
+        clear();
         dbmsProcessor.setupSharedDatabase();
         assertTrue(dbmsProcessor.checkBaseIntegrity());
-
-        clear(dbmsConnection);
     }
 
-    @ParameterizedTest
-    @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testInsertEntry(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
+    @Test
+    void testInsertEntry() throws SQLException {
         dbmsProcessor.setupSharedDatabase();
```

Comment:
It appears that every test first calls `setupSharedDatabase` so this method can be moved to the `setup` method.

Reply:
Besides `databaseIntegrityFullFiledAfterClearAndSetup` and `databaseIntegrityBrokenAfterClearedTables` - since we do not want to have so many different test classes, I accept that thE behavior might be different. Nevertheless, we have some tests :)